tot/models.py

```diff
@@ -225,3 +231,302 @@ def maybe_drop_added_dates(self, predicted, df):
+        # no installation checks required
+
+        # expect no _data_params
+        data_params = self.params["_data_params"]
+        assert len(data_params) == 0
+        custom_seasonalities = None  # TODO: auto-detect season from metadata
+
+        # Verify expected model_prams: K > 1, n_forecasts >1
+        model_params = deepcopy(self.params)
+        model_params.pop("_data_params")
+        assert len(model_params) == 2, "Model parameters K and n_forecasts must be provided"
+        self.n_forecasts = model_params["n_forecasts"]
+        assert self.n_forecasts > 1, "Model parameter n_forecasts must be >1. "
+        self.K = model_params["K"]  # for seasonal naive K is input parameter
+        assert self.K > 1, "Model parameter K must be >1 for seasonal naive. For K=1 select NaiveModel instead. "
+        self.n_lags = None  # TODO: should not be set to None. Find different solution.
+
+    def fit(self, df: pd.DataFrame, freq: str):
+        """Fits the naive model.
+        Naive models do not need to be explicitly fitted. However, we store fitting-related information.
+
+        Parameters
+        ----------
+            df : pd.DataFrame
+                dataframe containing column ``ds``, ``y``, and optionally ``ID`` with all data
+            freq : str
+                frequency of the input data
+        """
+        df, received_ID_col, received_single_time_series, received_dict, _ = df_utils.prep_or_copy_df(df)
+        # Receives df with single ID column. Only single time series accepted.
+        assert len(df["ID"].unique()) == 1  # TODO: add multi-ID, multi-target
+
+        self.freq = freq
+
+    def predict(self, df: pd.DataFrame):
+        """Runs the model to make predictions.
+        Expects all data to be present in dataframe.
+
+        Parameters
+        ----------
+            df : pd.DataFrame
+                dataframe containing column ``ds``, ``y``, and optionally ``ID`` with data
+        Returns
+        -------
+            pd.DataFrame
+                columns ``ds``, ``y``, optionally [``ID``], and [``yhat<i>``] where yhat<i> refers to the
+                i-step-ahead prediction for this row's datetime, e.g. yhat3 is the prediction for this datetime,
+                predicted 3 steps ago, "3 steps old".
+
+                Note
+                ----
+                 *  raw data is not supported
+        """
+        df, received_ID_col, received_single_time_series, received_dict, _ = df_utils.prep_or_copy_df(df)
+        # Receives df with single ID column. Only single time series accepted.
+        assert len(df["ID"].unique()) == 1  # TODO: add multi-ID, multi-target
+
+        forecast = pd.DataFrame
+        # check also no id column
+        for df_name, df_i in df.groupby("ID"):
+            dates, predicted = self._predict_raw(df_i)
+            forecast = self._reshape_raw_predictions_to_forecst_df(df_i, predicted)
+        fcst_df = df_utils.return_df_in_original_format(
```

Comment:
this may no longer be necessary now that dicts are no longer supported?

Reply:
Regarding the dicts I have the following proposal: I would rebuild the `df_utils` that we use from neuralprophet in a separate PR. When rebuilding e.g. the `neuralprophet.df_utils.prep_or_copy_df()`, I will drop the dict processing steps.